subsys/net/ip/dhcpv4.c

```diff
@@ -339,20 +339,11 @@ static void dhcpv4_send_request(struct net_if *iface)
 
 	iface->config.dhcpv4.attempts++;
 
-	const char *ciaddr_buf = 0;
-	static char pbuf[NET_IPV4_ADDR_LEN];
-
-	if (ciaddr) {
-		ciaddr_buf = net_addr_ntop(AF_INET, ciaddr, pbuf, sizeof(pbuf));
-	} else {
-		ciaddr_buf = "0.0.0.0";
-	}
-
 	NET_DBG("send request dst=%s xid=0x%"PRIx32" ciaddr=%s"
 		"%s%s timeout=%"PRIu32"s",
 		net_sprint_ipv4_addr(server_addr),
 		iface->config.dhcpv4.xid,
-		ciaddr_buf,
+		ciaddr ? net_sprint_ipv4_addr(ciaddr) : "0.0.0.0",
```

Comment:
Nak to this. The net_sprint_ipv4_addr() uses static buffer internally so there can be only one call to this in one NET_DBG()

Reply:
Indeed, I missed the fact it was already in use by server_addr.